packages/neos-ui-backend-connector/src/Endpoints/index.js

```diff
@@ -159,57 +174,80 @@ const parseGetSingleNodeResult = requestPromise => {
  */
-const getSingleNode = (nodeIdentifier, params = {}) => parseGetSingleNodeResult(fetch(urlWithParams('/neos/service/nodes/' + nodeIdentifier, params), {
+const getSingleNode = (nodeIdentifier, params = {}) => parseGetSingleNodeResult(fetchWithErrorHandling.withCsrfToken(() => ({
+    url: urlWithParams('/neos/service/nodes/' + nodeIdentifier, params),
+
     method: 'GET',
     credentials: 'include'
-}));
+})));
 
-const adoptNodeToOtherDimension = csrfToken => ({identifier, targetDimensions, sourceDimensions, workspaceName, copyContent = false}) => {
-    const params = {
+const adoptNodeToOtherDimension = ({identifier, targetDimensions, sourceDimensions, workspaceName, copyContent = false}) => parseGetSingleNodeResult(fetchWithErrorHandling.withCsrfToken(csrfToken => ({
+    url: '/neos/service/nodes',
+
+    method: 'POST',
+    credentials: 'include',
+    body: searchParams({
         identifier,
         dimensions: targetDimensions,
         sourceDimensions,
         workspaceName,
         mode: (copyContent ? 'adoptFromAnotherDimensionAndCopyContent' : 'adoptFromAnotherDimension'),
         __csrfToken: csrfToken
-    };
-
-    return parseGetSingleNodeResult(fetch('/neos/service/nodes', {
-        method: 'POST',
-        credentials: 'include',
-        body: searchParams(params)
-    }));
-};
+    })
+})));
 
-const setUserPreferences = csrfToken => (key, value) => {
+const setUserPreferences = (key, value) => fetchWithErrorHandling.withCsrfToken(csrfToken => {
     const data = new URLSearchParams();
     data.set('__csrfToken', csrfToken);
     data.set('key', key);
     data.set('value', value);
 
-    return fetch('neos/service/user-preferences', {
+    return {
+        url: 'neos/service/user-preferences',
+
         method: 'PUT',
         credentials: 'include',
         body: data
-    });
-};
+    };
+});
+
+const dataSource = (dataSourceIdentifier, dataSourceUri, params = {}) => fetchWithErrorHandling.withCsrfToken(() => ({
+    url: urlWithParams(dataSourceUri || '/neos/service/data-source/' + dataSourceIdentifier, params),
 
-const dataSource = (dataSourceIdentifier, dataSourceUri, params = {}) => fetchJson(urlWithParams(dataSourceUri || '/neos/service/data-source/' + dataSourceIdentifier, params), {
     method: 'GET',
     credentials: 'include'
-});
+})).then(response => response.json());
+
+const tryLogin = (username, password) => {
+    const data = new URLSearchParams();
+    data.set('__authentication[Neos][Flow][Security][Authentication][Token][UsernamePassword][username]', username);
+    data.set('__authentication[Neos][Flow][Security][Authentication][Token][UsernamePassword][password]', password);
+    // here, we
+    return fetch('/neos/login.json', {
+        method: 'POST',
+        body: data,
+        credentials: 'same-origin'
+    })
+    // parse the JSON if possible ...
+    .then(response => response.json())
```

Comment:
but won't it explode if json can't be parsed? Are you sure we don't need try/catch here?

Reply:
no, if json can't be parsed an exception will be raised, which will lead to the promise failure -- which is correct in this case :)